modules/services/window-managers/bspwm/default.nix

```diff
@@ -5,70 +5,72 @@ with lib;
 let
 
   cfg = config.xsession.windowManager.bspwm;
-  bspwm = cfg.package;
 
-  camelToSnake = s:
-    builtins.replaceStrings lib.upperChars (map (c: "_${c}") lib.lowerChars) s;
+  camelToSnake =
+    builtins.replaceStrings upperChars (map (c: "_${c}") lowerChars);
 
-  formatConfig = n: v:
+  formatMonitor = monitor: desktops:
+    "bspc monitor ${strings.escapeShellArg monitor} -d ${
+      strings.escapeShellArgs desktops
+    }";
+
+  formatSetting = n: v:
     let
-      formatList = x:
-        if isList x then
-          throw "can not convert 2-dimensional lists to bspwm format"
-        else
-          formatValue x;
-
-      formatValue = v:
-        if isBool v then
-          (if v then "true" else "false")
-        else if isList v then
-          concatMapStringsSep ", " formatList v
```

Comment:
lists are not supported anymore?

Reply:
I actually missed `ignore_ewmh_fullscreen` which seems to be the only option accepting a comma-separated list of values. I'll open a new PR.